4/m_recaptchav2.cpp

```diff
@@ -0,0 +1,292 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2015-2025 reverse Chevronnet
+ *   mike.chevronnet@gmail.com
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software; you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; version 2.
+ */
+
+/// $ModAuthor: reverse mike.chevronnet@gmail.com
+/// $ModConfig: mode="master/slave" masterserver="irc1.example.com" url="http://meme.com/verify/" message="*** reCAPTCHA: You need to verify that you are not a bot, to do so log in to Account or validate your connection here: {url}" 
+/// whitelistchans="#help,#opers" whitelistports="6667,6697">
+/// $ModDepends: core 4
+
+#include "inspircd.h"
+#include "modules/sql.h"
+#include "extension.h"
+#include "modules/account.h"
+#include "protocol.h"
+
+// Command to add or check an IP in the whitelist
+class ModuleCaptchaCheck;
+
+class CommandRecaptcha : public Command
+{
+private:
+    ModuleCaptchaCheck* parent;
+
+public:
+    CommandRecaptcha(Module* Creator, ModuleCaptchaCheck* Parent)
+        : Command(Creator, "RECAPTCHA", 2, 2), parent(Parent)
+    {
+        syntax = { "<add|check> <ip>" };
+    }
+
+    CmdResult Handle(User* user, const Params& parameters) override;
+};
+
+class ModuleCaptchaCheck : public Module
+{
+private:
+    std::string mode;
+    std::string captcha_url;
+    std::string masterserver;
+    SQL::Provider* sql = nullptr;
+    StringExtItem captcha_verified;
+    StringExtItem captcha_success;
+    CommandRecaptcha cmd;
+    Account::API accountapi;
+
+    friend class CommandRecaptcha;
+
+    class AddIPQuery : public SQL::Query
+    {
+    private:
+        ModuleCaptchaCheck* parent;
+        User* user;
```

Comment:
If the user disconnects before the query completes then this will result in a use after free. You should store the UUID In the class and then call `ServerInstance->Users.FindUUID` to look up the user.